src/redux/features/userSlice.ts

```diff
@@ -38,12 +44,25 @@ const userSlice = createSlice({
       (state, { payload }: PayloadAction<LoginResponse>) => {
         const token = payload.access
         state.token = token
+        state.verified = true
       }
     ),
       builder.addMatcher(
         authApi.endpoints.register.matchFulfilled,
         (state, { payload }: PayloadAction<LoginResponse>) => {
           state.token = payload.access
+          state.verified = false
+        }
+      ),
+      builder.addMatcher(
+        authApi.endpoints.verifyEmail.matchFulfilled,
+        (
+          state,
+          { payload }: PayloadAction<EmailVerificationResponseMessage>
+        ) => {
+          if (payload.message === 'Email verified successfully!') {
+            state.verified = true
```

Comment:
Refactor this, usually either cast the type of message have to be "Email verified successfully" or create a variable to hold the acceptable message, for example:

const payloadMessage = "Email verified successfully!"

Reply:
I decide not to use that method :)))